packages/rest/src/coercion/coerce-parameter.ts

```diff
@@ -60,56 +70,71 @@ export function coerceParameter(data: string, spec: ParameterObject) {
   }
 }
 
-/**
- * A set of truthy values. A data in this set will be coerced to `true`.
- *
- * @param data The raw data get from http request
- * @returns The corresponding coerced boolean type
- */
-function isTrue(data: string): boolean {
-  return ['true', '1'].includes(data);
+function coerceBuffer(data: string | object, spec: ParameterObject) {
+  if (typeof data === 'object')
+    throw RestHttpErrors.invalidData(data, spec.name);
+  return Buffer.from(data, 'base64');
 }
 
-/**
- * A set of falsy values. A data in this set will be coerced to `false`.
- * @param data The raw data get from http request
- * @returns The corresponding coerced boolean type
- */
-function isFalse(data: string): boolean {
-  return ['false', '0'].includes(data);
+function coerceDatetime(
+  data: string | object,
+  spec: ParameterObject,
+  dateOnly?: boolean,
+) {
+  if (typeof data === 'object' || isEmpty(data))
+    throw RestHttpErrors.invalidData(data, spec.name);
+
+  if (dateOnly) {
+    if (!matchDateFormat(data))
+      throw RestHttpErrors.invalidData(data, spec.name);
+  } else {
+    if (!isRFC3339(data)) throw RestHttpErrors.invalidData(data, spec.name);
+  }
+
+  const coercedDate = new Date(data);
+  if (!isValidDateTime(coercedDate))
+    throw RestHttpErrors.invalidData(data, spec.name);
+  return coercedDate;
 }
 
-/**
- * Return the corresponding OpenAPI data type given an OpenAPI schema
- *
- * @param type The type in an OpenAPI schema specification
- * @param format The format in an OpenAPI schema specification
- */
-function getOAIPrimitiveType(type?: string, format?: string) {
-  // serizlize will be supported in next PR
-  if (type === 'object' || type === 'array') return 'serialize';
-  if (type === 'string') {
-    switch (format) {
-      case 'byte':
-        return 'byte';
-      case 'binary':
-        return 'binary';
-      case 'date':
-        return 'date';
-      case 'date-time':
-        return 'date-time';
-      case 'password':
-        return 'password';
-      default:
-        return 'string';
-    }
+function coerceNumber(data: string | object, spec: ParameterObject) {
+  if (typeof data === 'object' || isEmpty(data))
+    throw RestHttpErrors.invalidData(data, spec.name);
+
+  const coercedNum = Number(data);
+  if (isNaN(coercedNum)) throw RestHttpErrors.invalidData(data, spec.name);
+
+  debug('data of type number is coerced to %s', coercedNum);
+  return coercedNum;
+}
+
+function coerceInteger(
+  data: string | object,
+  spec: ParameterObject,
+  isLong?: boolean,
```

Comment:
Boolean arguments are considered a bad practice.

Proposed alternatives:
 - convert this last argument to `options: {isLong: boolean}` 
 - compute `isLong` inside `coerceInteger` as `isLong = spec.type === 'long'`. I personally prefer this option.